test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs

```diff
@@ -570,17 +570,31 @@ public void DefaultContentRootIsApplicationBasePath()
         }
 
         [Fact]
-        public void DefaultApplicationNameWithNoStartupThrows()
+        public void DefaultWebHostBuilderWithNoStartupThrows()
         {
-            var builder = new ConfigurationBuilder();
             var host = new WebHostBuilder()
                 .UseServer(new TestServer());
 
-            var ex = Assert.Throws<ArgumentException>(() => host.Build());
+            var ex = Assert.Throws<InvalidOperationException>(() => host.Build());
+
+            Assert.Equal("No service for type 'Microsoft.AspNetCore.Hosting.IStartup' has been registered.", ex.Message);
+        }
+
+        [Fact]
+        public void DefaultApplicationNameToStartupAssemblyName()
+        {
+            var builder = new WebHostBuilder()
+                .ConfigureServices(services =>
+                {
+                    services.AddSingleton<IStartup, Startup>();
+                })
+                .UseServer(new TestServer());
 
-            // ArgumentException adds "Parameter name" to the message and this is the cleanest way to make sure we get the right
-            // expected string
-            Assert.Equal(new ArgumentException("A valid non-empty application name must be provided.", "applicationName").Message , ex.Message);
+            using (var host = builder.Build())
+            {
+                var hostingEnv = host.Services.GetService<IHostingEnvironment>();
+                Assert.Equal(Assembly.GetEntryAssembly()?.GetName().Name, hostingEnv.ApplicationName);
```

Comment:
I'm going to change this to a functional test so the actual assembly name is resolved instead of the xunit's testhost.